src/content/organizations/ListView.tsx

```diff
@@ -294,25 +287,82 @@ const ListingView: FC = () => {
         );
 
         return nameMatch || abbrMatch;
-      })
-      .sort((a, b) => orderBy?.comparator(a, b) || 0);
+      },
+    ];
+
+    const filteredData = data.filter((u) => filters.every((filter) => filter(u)));
+    const sortedData = sortData(filteredData, orderBy, sortDirection, comparator);
+    const paginatedData = sortedData.slice(offset, first + offset);
+
+    setCount(sortedData?.length);
+    setDataset(paginatedData);
+  };
 
-    if (order === "desc") {
-      sorted.reverse();
+  const isStatusFilterOption = (status: string): status is FilterForm["status"] =>
+    ["All", "Inactive", "Active"].includes(status);
+
+  useEffect(() => {
+    if (!data?.length) {
+      return;
     }
 
-    setCount(sorted.length);
-    setDataset(sorted.slice(page * perPage, page * perPage + perPage));
-  }, [data, perPage, page, orderBy, order, studyFilter, orgFilter, statusFilter]);
+    const organizationId = searchParams.get("organization") || "";
+    const study = searchParams.get("study") || "";
+    const status = searchParams.get("status");
+
+    if (organizationId !== orgFilter) {
+      setValue("organization", organizationId);
+    }
+    if (study !== studyFilter) {
+      setValue("study", study);
+    }
+    if (isStatusFilterOption(status) && status !== statusFilter) {
```

Comment:
If you set the query parameter manually in the URL to an invalid option (e.g. `status=XXXX`), the Status selectbox shows up as empty instead of the default option (`All` in this case). It also shows no data in the table, probably because it's checking for an organization with status `XXXX`.

This doesn't happen for the User List, so I'm mentioning it incase we're missing some validation logic. If the resolution is high effort, we can skip addressing this.

Reply:
Was due to the way I was setting the default values for the filters. Made fix.